### DISCUSSION TITLE
I can not find some document?

### DISCUSSION THREAD

**Discussion by ruyi-tsai** [microsoft/jacdac#873] [Q&A] [answered]
I read https://microsoft.github.io/jacdac-docs/reference/protocol 
I cannot find system specification and control service specification document ? could you send me? or find give me some link?
in Error detection section 
The image and table below captures key timings of Jacdac SWS.
below table discuss 	Duration min/max (us)
I think Duration min/max (ms)?

**Comment by ruyi-tsai** [comment]
The image and table below captures key timings of Jacdac SWS.
below table discuss  Duration min/max (us)
I think Duration min/max (ms)? => this problem cancel

**Comment by jamesadevine** [comment, marked as answer]
The control service specification is here: https://microsoft.github.io/jacdac-docs/services/control/

Here is a [client implementation](https://github.com/microsoft/pxt-jacdac/blob/master/routing.ts) and here is a [server implementation](https://github.com/microsoft/jacdac-c/blob/0b59bf72237855dfe803dba24f581fdb86666b63/source/jd_control.c).
